tools/src/main/java/org/apache/kafka/tools/consumergroup/ConsumerGroupCommandOptions.java

```diff
@@ -0,0 +1,263 @@
+            if (!options.has(groupOpt) || !options.has(topicOpt))
+                CommandLineUtils.printUsageAndExit(parser,
+                    "Option " + deleteOffsetsOpt + " takes the following options: " + Utils.join(allDeleteOffsetsOpts));
+        }
+
+        if (options.has(resetOffsetsOpt)) {
+            if (options.has(dryRunOpt) && options.has(executeOpt))
+                CommandLineUtils.printUsageAndExit(parser, "Option " + resetOffsetsOpt + " only accepts one of " + executeOpt + " and " + dryRunOpt);
+
+            if (!options.has(dryRunOpt) && !options.has(executeOpt)) {
+                System.err.println("WARN: No action will be performed as the --execute option is missing." +
+                    "In a future major release, the default behavior of this command will be to prompt the user before " +
+                    "executing the reset rather than doing a dry run. You should add the --dry-run option explicitly " +
+                    "if you are scripting this command and want to keep the current default behavior without prompting.");
+            }
+
+            if (!options.has(groupOpt) && !options.has(allGroupsOpt))
+                CommandLineUtils.printUsageAndExit(parser,
+                    "Option " + resetOffsetsOpt + " takes one of these options: " + Utils.join(allGroupSelectionScopeOpts));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetToOffsetOpt,   minus(allResetOffsetScenarioOpts, resetToOffsetOpt));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetToDatetimeOpt, minus(allResetOffsetScenarioOpts, resetToDatetimeOpt));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetByDurationOpt, minus(allResetOffsetScenarioOpts, resetByDurationOpt));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetToEarliestOpt, minus(allResetOffsetScenarioOpts, resetToEarliestOpt));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetToLatestOpt,   minus(allResetOffsetScenarioOpts, resetToLatestOpt));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetToCurrentOpt,  minus(allResetOffsetScenarioOpts, resetToCurrentOpt));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetShiftByOpt,    minus(allResetOffsetScenarioOpts, resetShiftByOpt));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetFromFileOpt,   minus(allResetOffsetScenarioOpts, resetFromFileOpt));
+        }
+
+        CommandLineUtils.checkInvalidArgs(parser, options, groupOpt, minus(allGroupSelectionScopeOpts, groupOpt));
+        CommandLineUtils.checkInvalidArgs(parser, options, groupOpt, minus(allConsumerGroupLevelOpts, describeOpt, deleteOpt, resetOffsetsOpt));
+        CommandLineUtils.checkInvalidArgs(parser, options, topicOpt, minus(allConsumerGroupLevelOpts, deleteOpt, resetOffsetsOpt));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> Set<T> minus(Set<T> set, T...toRemove) {
```

Comment:
What about moving this method to `CollectionUtils`?

Reply:
Yes maybe we can move this to Utils?

Reply:
Moved.